### ISSUE TITLE
Mob grinder: No drops *only* from chickens??

### ISSUE THREAD

**Issue by djcprone** [brandon3055/Draconic-Evolution#1362]
So basically, I have a large mob spawner set up. it works flawlessly for 12 mobs. The draconic mob grinder kills withers, creepers, pigs, cows, squid, slimes, ghasts, etc.- instantly and without fail and all the items drop along with the xp. Cool. But! If you throw chickens in front of a mob grinder, it kills it instantly as it should, but zero drops of any kind. No matter what, i let it go for an hour with no drops. Turn off the mob grinder and kill it myself, or use an MFR grinder and I get the drops as usual. I don't know programming so I really don't know what kinda weird thing would cause that but it is literally only chickens. Now, the problem is, I could drop MFR grinders here and there around the spawner to kill the chickens but because your mob grinder is so good and fast, it kills them faster than the idle-based MFR unit can, so I still get no drops. Anything I can do to change this? or is there a patched version for 1.10.2 that just isn't in the current All the Mods release?

Thanks for your time. I did search open and closed tickets but I didn't see anything like this. And the issue being so specific to chickens made a google search pretty much useless.

TLDR: mob grinder kills everything perfectly for me x12, but when it kills chickens you get no drops no matter what.

## Base information
* Minecraft version: 1.10.2
* Mod version: Draconic-Evolution-1.10.2-2.1.8.273-universal
* Minecraft Forge version: forge-1.10.2-12.18.3.2511-universal
* Mod Pack: All the Mods (v4.2)

There is no crash report- no crash, just seems to be a weird bug or exception? I can provide any further info you need so please don't insta close me :)

**Comment by FoxMcloud5655** [comment]
Unfortunately, versions below MC 1.12.2 are not updated anymore.  Sorry about that.

**Comment by djcprone** [comment]
is it possible for me to edit or change anything on an individual basis to
make it work on my server? I want to change to 1.13 or something but
everyone on the server is so developed I don't want to abandon it yet

**Comment by FoxMcloud5655** [comment]
You can absolutely take the 1.7.10 branch on this repository and edit it however you like for your own usage.

I understand not wanting to change to a higher version, but there are many, MANY mods not supporting 1.7.10 anymore, and once you get used to the combat system, 1.12.2 just feels right after a while.

**Comment by djcprone** [comment]
I loved 1.7.10. I'm on 1.10.2 now and tbh I hate it. They added some cool
features but overall it's laggy and riddled with bugs compared to 1.7.10
and the chunk loading is horrendous. I think I'll upgrade at the beginning
of the new year. How hard would it be for me to edit the repository just to
get drops from chickens? Like that's all I'm missing as far as I can tell.
I've ran probably 25 different mobs through the mob grinder with no issues.
Just chickens seem to have been overlooked. And is that something I only
need to edit server side? Or would I need to send that file then to each
client as well?

Thanks for any help you can give me.

**Comment by FoxMcloud5655** [comment]
Sorry, I didn't see that you were on 1.10.2.  My fault.  There's pretty much no reason for you not to update to 1.12.2, as most mods will be supported if you do upgrade.  1.10.2 is not great; you're right.  1.12.2 fixes quite a few things I didn't like about 1.8 and 1.10.

If you did edit the 1.10.2 branch, then I would send your updated version to all clients once it's finished, even if it's only a server-side change.

**Comment by djcprone** [comment]
Mostly just stuck in it because of the people losing everything they've
built. I assume if I went from All the Mods 1 on 1.10.2, to All the Mods 3
on 1.12.2, I'd have to do a whole new world and everyone starts over. But
the chunkloading is just ridiculous in 1.10.2. I'm running on an i7 4700,
32gb ram (12gb to server on dedicated 1tb ssd) with wired true gigabit
internet averaging 960MBps down and 880MBps up. Like there's no reason
literally ever that anyone on my server can't fly around and chunk load
without total lag and rubber banding. Wasn't an issue on 1.7.10 for us at
all.

I'll have to look into how to edit that file because I have no idea what
controls getting drops from the draconic grinder at all lol

**Comment by FoxMcloud5655** [comment]
If you're having lag like that, try pre-genning all of the chunks within your world border.  If you don't have a world border set up, I also suggest doing that too.  It will help immensely with loading times.

As for the drops, I have no idea what could be causing them not to drop.  At least in 1.12.2 and as of  the time of this post, the code for it is as follows:
```java
fakePlayer = FakePlayerFactory.get((WorldServer) world, new GameProfile(UUID.fromString("5b5689b9-e43d-4282-a42a-dc916f3616b7"), "[Draconic-Evolution]"));
DamageSource source = DamageSource.causePlayerDamage(fakePlayer);
damage = Float.MAX_VALUE;
target.attackEntityFrom(source, damage);
```
The Mob Grinder doesn't actually handle any drops; it only attacks the target to kill it, letting MC handle the drops.
